src/main/java/net/shortninja/staffplus/core/common/cmd/CommandPlayerRetriever.java

```diff
@@ -28,9 +28,14 @@ public Optional<SppPlayer> retrievePlayer(PlayerRetrievalStrategy strategy, Stri
         if (strategy == NONE) {
             return Optional.empty();
         }
-
-        if ((strategy == OPTIONAL_BOTH || strategy == OPTIONAL_ONLINE) && playerName == null) {
-            return Optional.empty();
+        
+        if (playerName == null) {
+            if (strategy == OPTIONAL_BOTH || strategy == OPTIONAL_ONLINE) {
+                return Optional.empty();
+            }
+            
+            // User of this method inputed null
+            return null;

```

Comment:
I don't think returning null here is correct? Shouldn't the method just return optional.empty? 
I believe we can just throw an error here when the playername is null and the strategy is ONLINE or BOTH